src/openklant/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,176 @@
+)
+
+PARTIJEN_PATH = reverse_lazy(
+    "klantinteracties:partij-list",
+    kwargs=dict(version=settings.REST_FRAMEWORK["DEFAULT_VERSION"]),
+)
+
+
+def _retrieve_klanten(url: str, access_token: str) -> Tuple[list[Klant], str | None]:
+    klanten_path = "/klanten/api/v1/klanten"
+
+    _url = urlparse(url)
+    _params = parse_qs(_url.query)
+
+    client = LegacyOpenKlantClient(f"{_url.scheme}://{_url.netloc}", access_token)
+    response_data = client.retrieve(_url.path or klanten_path, params=_params)
+
+    if not isinstance(response_data, dict):
+        logger.error(f"Unexpected response data returned: {response_data}")
+        return [], None
+
+    items = response_data.get("results", [])
+    klanten = []
+
+    generic_client = Client()
+
+    for data in items:
+        klant = Klant(
+            **{field: value for field, value in data.items() if field in KLANT_FIELDS}
+        )
+
+        if klant.subject and not klant.subject_type:
+            subject_data = generic_client.retrieve(klant.subject)
+
+            if not isinstance(subject_data, dict):
+                logger.error(
+                    "Unexpected response data returned during retrieval of "
+                    f"subject: {subject_data}"
+                )
+
+                continue
+
+            klant.set_from_external_subject(subject_data)
+
+        klanten.append(klant)
+
+    next_url = response_data.get("next")
+    return klanten, next_url
+
+
+def _save_klanten(url: str, klanten: list[Klant]) -> list[str]:
+    if not klanten:
+        logger.info("No klanten to save, moving on...")
+        return []
+
+    logger.info(f"Trying to create {len(klanten)} klanten through the V2 API")
+
+    dummy_token = _generate_dummy_token()
+    openklant_client = OpenKlantClient(url, dummy_token)
+    created_klanten = []
+
+    for klant in klanten:
+        digitaal_adres = klant.to_digitaal_adres()
+        digitaal_adres_ref = None
+
+        if digitaal_adres:
+            _data = openklant_client.create(
+                DIGITALE_ADDRESSEN_PATH, digitaal_adres.dict()
+            )
+
+            if not isinstance(_data, dict):
+                logger.error(
+                    f"Unknown data received creating a digitaal adres: {_data}. "
+                    "Skipping klant."
+                )
+                continue
+
+            digitaal_adres_ref = _data.get("uuid")
+
+        partij = klant.to_partij(digitaal_adres=digitaal_adres_ref)
+
+        if not partij:
+            logger.error(f"Unable to create partij for klant: {asdict(klant)}")
+            continue
+
+        response_data = openklant_client.create(PARTIJEN_PATH, partij.dict())
+
+        if response_data and "url" in response_data:
+            created_klanten.append(response_data["url"])
+
+    return created_klanten
+
+
+def _generate_dummy_token() -> str:
+    token_auth, _ = TokenAuth.objects.get_or_create(application="Migration application")
```

Comment:
It might be good to remove this token after running the command (and also if the command fails)

Reply:
See https://github.com/maykinmedia/open-klant/pull/263/commits/293156ee77fa8ee876d8d0d92053eaf9e6cd1c7f